src/EndcapTOF_geo.cpp

```diff
@@ -184,13 +183,12 @@ static Ref_t create_detector(Detector& description, xml_h e, SensitiveDetector s
 
   // NOTE: ACTS extension for the disk layer of the TTL
   // also defining the coordinate system that differs between ACTS and Geant4 (zyx vs xyz)
-  Acts::ActsExtension* detlayer = new Acts::ActsExtension();
-  detlayer->addValue(-80. * mm, "r_min", "envelope");
-  detlayer->addValue(670. * mm, "r_max", "envelope");
-  detlayer->addValue(10. * mm, "z_min", "envelope");
-  detlayer->addValue(10. * mm, "z_max", "envelope");
-  detlayer->addType("sensitive plane", "layer");
-  layer_detEl.addExtension<Acts::ActsExtension>(detlayer);
+  auto &layerParams = DD4hepDetectorHelper::ensureExtension<dd4hep::rec::VariantParameters>(
+                      layer_detEl);
+  layerParams.set<double>("envelope_r_min", -80. * mm);
+  layerParams.set<double>("envelope_r_max", 670. * mm);
+  layerParams.set<double>("envelope_z_min", 10. * mm);
+  layerParams.set<double>("envelope_z_max", 10. * mm);
```

Comment:
Can/should we put this into the compact file?

Reply:
I have for now put this [here](https://github.com/eic/epic/blob/34-refactor-the-dd4hep-acts-integration-for-acts-20-0-0-and-dd4hep-1-21/compact/tracking/tof_endcap.xml#L146) and parsed [here](https://github.com/eic/epic/blob/34-refactor-the-dd4hep-acts-integration-for-acts-20-0-0-and-dd4hep-1-21/src/EndcapTOF_geo.cpp#L54). But this is not ideal. The intended parametrization is not clear to me. I will ask nicholas how they were derived. There is also another envelope tag defined in the compact file which is never parsed. Should sort this in a separate PR like the MPGD.